enumerable.rb

```diff
@@ -0,0 +1,179 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr = to_a if instance_of?(Range)
+    arr = flatten if instance_of?(Hash)
+
+    count = 0
+    while count < arr.length
+      yield(arr[count])
+      count += 1
+    end
+    arr
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr = to_a if instance_of?(Range)
+    arr = flatten if instance_of?(Hash)
+
+    count = 0
+    while count < arr.length
+      yield(arr[count], count)
+      count += 1
+    end
+    arr
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    my_each { |it| new_arr << it if yield(it) }
+    new_arr
+  end
+
+  def my_all?(param = nil)
```

Comment:
enumerables #my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Numeric
 ```
     Failure/Error: expect(numbers.my_all?(Numeric)).to be numbers.all?(Numeric)
     
       expected true
            got false
``` 

enumerables #my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex

```     Failure/Error: expect(words.my_all?(/d/)).to be words.all?(/d/)
     
```

enumerables #my_all? when a Regex is passed as an argument returns false if any of the collection does not match the Regex
```
     Failure/Error: expect(words.my_all?(/o/)).to be words.all?(/o/)
```     

  
  enumerables #my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

```    
 Failure/Error: expect(array.my_all?(3)).to be array.all?(3)
```     

enumerables #my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern
```
     Failure/Error: expect(array.my_all?(3)).to be array.all?(3)
```
#
### `my_all?` should be identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2)).